### ISSUE TITLE
Feature logic generates nextPageToken at wrong layer

### ISSUE THREAD

**Issue by dcolligan** [ga4gh/server#1288]
The features iterator in sequenceAnnotations.py:Gff3DbFeatureSet.getFeatures returns a (feature, nextPageToken) tuple ... it has the nextPageToken logic at the wrong layer of abstraction

Revise the tests in tests/unit/test_data_interface.py appropriately 

Discovered in #1219

**Comment by jeromekelleher** [comment]
I wouldn't worry too much about fixing the layering --- the paging logic is being shoved down the stack in the SQL Alchemy refactor. If you see access patterns that are different to the rest of the API please flag them with comments and we'll hopefully make everything more uniform once the refactor has been completed.

**Comment by david4096** [comment]
@jeromekelleher You should consider using SQL alchemy for the sequence annotations backing, if you are not. I don't believe there's anything that wouldn't be done better that way, instead of our own sqlite backing.

**Comment by jeromekelleher** [comment]
Yeah, I agree @david4096. I think we should do this once we've changed over the main registry code to the SQL alchemy stuff in master. We can also think about storing other stuff (like variants) in a SQL DB.

**Comment by david4096** [comment]
I think this might need to be fixed sooner as I'm getting a 500 trying to page through results on 1kgenomes now.

```[Mon Aug 01 13:48:28.729620 2016] [:error] [pid 45174:tid 139763961530112] Exception on / [GET]
[Mon Aug 01 13:48:28.729660 2016] [:error] [pid 45174:tid 139763961530112] Traceback (most recent call last):
[Mon Aug 01 13:48:28.729664 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
[Mon Aug 01 13:48:28.729667 2016] [:error] [pid 45174:tid 139763961530112]     rv = self.dispatch_request()
[Mon Aug 01 13:48:28.729670 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
[Mon Aug 01 13:48:28.729673 2016] [:error] [pid 45174:tid 139763961530112]     return self.view_functions[rule.endpoint](**req.view_args)
[Mon Aug 01 13:48:28.729675 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/frontend.py", line 577, in searchFeatures
[Mon Aug 01 13:48:28.729678 2016] [:error] [pid 45174:tid 139763961530112]     flask.request, app.backend.runSearchFeatures)
[Mon Aug 01 13:48:28.729681 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/frontend.py", line 440, in handleFlaskPostRequest
[Mon Aug 01 13:48:28.729683 2016] [:error] [pid 45174:tid 139763961530112]     return handleHttpPost(flaskRequest, endpoint)
[Mon Aug 01 13:48:28.729686 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/frontend.py", line 315, in handleHttpPost
[Mon Aug 01 13:48:28.729688 2016] [:error] [pid 45174:tid 139763961530112]     responseStr = endpoint(request.get_data())
[Mon Aug 01 13:48:28.729691 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/backend.py", line 1076, in runSearchFeatures
[Mon Aug 01 13:48:28.729693 2016] [:error] [pid 45174:tid 139763961530112]     self.featuresGenerator)
[Mon Aug 01 13:48:28.729696 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/backend.py", line 776, in runSearchRequest
[Mon Aug 01 13:48:28.729698 2016] [:error] [pid 45174:tid 139763961530112]     for obj, nextPageToken in objectGenerator(request):
[Mon Aug 01 13:48:28.729701 2016] [:error] [pid 45174:tid 139763961530112]   File "/srv/ga4gh/ga4gh-protobuf/lib/python2.7/site-packages/ga4gh-0.2.3.dev269+ngbb55ded-py2.7.egg/ga4gh/datamodel/sequenceAnnotations.py", line 485, in getFeatures
[Mon Aug 01 13:48:28.729704 2016] [:error] [pid 45174:tid 139763961530112]     nextPageToken += 1
[Mon Aug 01 13:48:28.729707 2016] [:error] [pid 45174:tid 139763961530112] TypeError: unsupported operand type(s) for +=: 'NoneType' and 'int'```

**Comment by dcolligan** [comment]
Fixed in #1370
